### DISCUSSION TITLE
First alpha release of PowerDNS DNSdist 1.9.0

### DISCUSSION THREAD

**Discussion by rgacogne** [PowerDNS/pdns#13270] [Announcements]
https://blog.powerdns.com/first-alpha-release-of-dnsdist-1.9.0

**Comment by phonedph1** [comment]
![image](https://github.com/PowerDNS/pdns/assets/20867105/46882baf-766c-4580-afab-a0e5ee2f3de8)
![image](https://github.com/PowerDNS/pdns/assets/20867105/6a31f989-83fd-4ba9-834b-55b44ee6a55e)

I *think* this is user error so I am asking here first :)

Did any of the connection timeout handling in nghttp2 change drastically compared to h2o?

It looks like idle connections get a closed timeout after ~3 seconds compared to the 30s default under h2o. 

At a small scale this looks like much more new connections, much less active connections.

Running Firefox to a DoH endpoint we get:

```
Sep 25 16:48:18 Got answer from 127.0.0.1:5353, relayed to <me>:35358 (DoH, 0 bytes), took 20679.1 us
Sep 25 16:48:20 Timeout (read) from remote H2 client <me>:35358
Sep 25 16:48:20 Timeout while reading from TCP client <me>:35358

Sep 25 16:48:25 Got answer from 127.0.0.1:5353, relayed to <me>:43418 (DoH, 0 bytes), took 491.887 us
Sep 25 16:48:28 Timeout (read) from remote H2 client <me>:43418
Sep 25 16:48:28 Timeout while reading from TCP client <me>:43418

Sep 25 16:48:37 Got answer from 127.0.0.1:5353, relayed to <me>:46806 (DoH, 0 bytes), took 532.225 us
Sep 25 16:48:40 Timeout (read) from remote H2 client <me>:46806
Sep 25 16:48:40 Timeout while reading from TCP client <me>:46806

Sep 25 16:48:12 Got answer from 127.0.0.1:5353, relayed to <me>:55798 (DoH, 0 bytes), took 249.591 us
Sep 25 16:48:15 Timeout (read) from remote H2 client <me>:55798
Sep 25 16:48:15 Timeout while reading from TCP client <me>:55798
```

Setting `setTCPRecvTimeout(30)` seems to make it act like normal.... so perhaps this is just something that should be noted? Or is there a separate tuning now applicable to nghttp2?

**Reply by rgacogne** [reply]
Thank for reporting this! It looks like a bug to me, because the documentation states that `addDOHLocal`'s `idleTimeout` should still be honored when `nghttp2` is used. I'm looking into it!

**Reply by rgacogne** [reply]
For the record, it should be fixed in alpha2 (not released yet) by https://github.com/PowerDNS/pdns/pull/13298
